### PR TITLE
Update parse_protobuf_file_scan_config to remove any partition columns from the file_schema in FileScanConfig

### PR DESCRIPTION
**Pull request by bcmcmill**
## Which issue does this PR close?

Closes https://github.com/apache/arrow-ballista/issues/747.
Closes https://github.com/apache/arrow-ballista/issues/862.

## Rationale for this change

This change corrects an issue in `arrow-ballista` causing partitioned ListingTables fail when being read.

## What changes are included in this PR?

A simple change to the `parse_protobuf_file_scan_config` function to filter the table_partition_cols out of the file_schema.

## Are these changes tested?

All current tests cases pass and the change corrects the issue in `arrow-ballista`

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/proto/Cargo.toml [hunk 1]
```diff
@@ -56,4 +56,4 @@ serde_json = { workspace = true, optional = true }
 [dev-dependencies]
 doc-comment = { workspace = true }
 strum = { version = "0.26.1", features = ["derive"] }
-tokio = "1.18"
+tokio = { version = "1.18", features = ["rt-multi-thread"] }
```
Comment (alamb) [inline datafusion/proto/Cargo.toml:59]: is this change needed?

File: datafusion/proto/src/physical_plan/from_proto.rs [hunk 2]
```diff
@@ -560,9 +560,21 @@ pub fn parse_protobuf_file_scan_config(
         output_ordering.push(sort_expr);
     }
 
+    // Currently when the schema for the file is set the partition columns
+    // are present, which is illegal because they aren't actually in the files.
+    // This is a workaround to remove them from the schema.
+    let file_schema = Arc::new(Schema::new(
```
Comment (alamb) [inline datafusion/proto/src/physical_plan/from_proto.rs:566]: I don't undersrtand why the protobuf code would be doing this filtering -- wouldn't this potentially mask an error in whatever was creating the protobuf?

Perhaps it would be better to apply this filtering prior to serializing the plan?

If this is needed in the protobuf serialization code, can you please add a test to 
1. Show how this happens and 
2. Ensure we don't accidentally break this functionality in a future refactor?

**Review by alamb — COMMENTED** [review]
Thank you for this contribution @bcmcmill  -- I don't fully understand the proposal though I did not read through to the ballista issues

File: datafusion/proto/src/physical_plan/from_proto.rs (hunk 2 above)
Comment (bcmcmill) [reply datafusion/proto/src/physical_plan/from_proto.rs:566]: Absolutely.

Comment (bcmcmill) [reply datafusion/proto/src/physical_plan/from_proto.rs]: These are complete.

File: datafusion/proto/Cargo.toml (hunk 1 above)
Comment (bcmcmill) [reply datafusion/proto/Cargo.toml:59]: ```error: The default runtime flavor is `multi_thread`, but the `rt-multi-thread` feature is disabled.
  --> datafusion/proto/examples/logical_plan_serde.rs:22:1
   |
22 | #[tokio::main]
   | ^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `tokio::main` (in Nightly builds, run with -Z macro-backtrace for more info)```
   
The above error is hit when `rt-multi-thread` is absent.

File: datafusion/proto/tests/cases/roundtrip_physical_plan.rs [hunk 3]
```diff
@@ -561,6 +563,32 @@ fn roundtrip_parquet_exec_with_pruning_predicate() -> Result<()> {
     )))
 }
 
+#[tokio::test]
+async fn roundtrip_parquet_exec_with_table_partition_cols() -> Result<()> {
```
Comment (alamb) [inline datafusion/proto/tests/cases/roundtrip_physical_plan.rs:567]: I verified this test fails without the code changes in this file. 👍 

```

from proto: ArrowError(SchemaError("Unable to get field named \"part\". Valid fields: [\"col\"]"), None)
thread 'cases::roundtrip_physical_plan::roundtrip_parquet_exec_with_table_partition_cols' panicked at datafusion/proto/tests/cases/roundtrip_physical_plan.rs:111:10:
from proto: ArrowError(SchemaError("Unable to get field named \"part\". Valid fields: [\"col\"]"), None)
stack backtrace:
```

**Review by alamb — APPROVED** [review]
Looks good to me -- thanks again @bcmcmill

**Comment by alamb** [comment]
Thanks again @bcmcmill
